put your name, etc here

## Paola Granados 
1/19/2021

# Lecture 1.2

Short lecture today, and mostly logistics. I will introduce the final projects and grading next week. (Sorry!) Short story for the grading is that you have to complete these first labs to pass the class, and I am grading them in a binary done/not-done scheme. So you need to complete them.

Lecture Agenda:

* debriefing on first class and homework, how's it going?
* logistics of github, downloading this Lab 1.2 notebook
* commenting and markup in notebooks
* workflow for the class


In general, we will do warmup exercises and lab exercises in a jupyter notebook. For most classes, you will download the notebook from github, open it on your local computer, and then upload your completed work back on github in  your folder. In general, all lab work for the week is due Sunday before midnight on blackboard.




In [ ]:
# here is an example of code commenting. The # symbol means everything on this line will be ignored 




Here is an example of markdown text. Because this cell is in markdown mode it will also be ignored by the python shell

## Preliminaries

In our class github site, create a folder for yourself where you will submit your jupyter notebooks. The easiest way to do this is to create a new file that is a placeholder document in the new folder. You can delete it later.

## Warmup exercises

As you are doing these exercises, get familiar with jupyter notebook commands 
https://towardsdatascience.com/jypyter-notebook-shortcuts-bf0101a98330

and markdown text https://medium.com/@ingeh/markdown-for-jupyter-notebooks-cheatsheet-386c05aeebed

These exercises are based on what you have learned previously. Feel free to go look at what you did in datacamp to figure out how to answer them.

**W.1** Define and assign values to three variables of the three types we learned about Monday: integer, floating point, string and boolian. 

Note you can add more cells below by pressing the + button above or typing [esc] to command mode, then [b] to enter a cell below, or [a] to enter a cell above. See the link above for more commands

In [6]:
#integer 
a = 7

#float
b = 3.1415

#string 
c = "Paola"

#Boolean 
d = False

**W.2** Print out the value and check the type of each variable

In [7]:
#check for integer
print(a)
type(a)

7


int

In [8]:
#check for float
print(b)
type(b)

3.1415


float

In [9]:
#check for string
print(c)
type(c)

Paola


str

In [10]:
#check for boolean
print(d)
type(d)

False


bool

**W.3** Create a new variable that saves the value of the float variable as a string. Print it out

In [11]:
b_str = str(b)

In [12]:
print(b_str)
type(b_str)

3.1415


str

**W.4** Using multiplication of a variable, print out the text "Bueno Bueno Bueno Bueno Bueno" (Shoutout to Monsieur Periné https://youtu.be/JGL-eQAAxGs see 10:45)

In [13]:
#make the word Bueno a string variable
x = "Bueno "

In [14]:
#multiply the variable
x1 = x*5
print(x1)

Bueno Bueno Bueno Bueno Bueno 


**W.5** Do the same thing using addtion

In [15]:
#add the variable 
x2 = x+x+x+x+x
print(x2)

Bueno Bueno Bueno Bueno Bueno 


**W.6** If you add an int and a float, what is the resulting variable type? Demonstrate

In [16]:
#adding the inital float b and integer a
y = b + a
type(y)

float

    Adding an int and a float results in a float 

**W.7** Can you convert a Boolian to an interger type variable? If so, what is the resulting value? What happens if you convert a float to an int?

In [17]:
# converting Boolean d to an Integer 
I1 = int(d)
print(I1)

0


In [18]:
e = True
I2 = int(e)
print(I2)

1


    Converting a boolean to integer results to 0 if false, and 1 if true

In [19]:
#Converting float to an int
I2 = int(b)
print(I2)

3


    Converting a float to an int rounds the number, in this case 3.14 was rounded to 3

# Lab 1.2

## Lab Exercises

**E.1** Complete Ch 2 of Introduction to Python in datacamp. 

**E.2** Below you will spend time practicing what you have learned in Ch 2. Use a comination of code, code commenting, and markdown text to take notes. Make sure you investigate how python syntax works and understand what you are doing, and what went wrong if you have any issues. You can refer to this document later if you need to review, so give future you lots of commenting to be helpful.

**E.2.1** Let's work on lists, indexing and printing out elements of lists.

We are going to calculate the BMI of five family members (or friends/colleaues if you have a small family, or famous people/politicians/atheletes if you really don't like talking to people, they just have to be adults and you need their data). Make intellegently named lists for the first name, last name, height, weight and age of each person. So, five lists in total.

In [20]:
#Lists of family members' information 
# height in in
# weight in lb
firstname = ["Marcela","Margoth","Juan","Karina","Feluel"]
lastname = ["Granados","Barrios","Granados","Barrios","Barrios"]
height = [65,67,70,64,73]
weight = [130,150,230,160,215]
age = [18,44,47,40,42]


**E.2.2** Now create a new list that calculates the BMI of each person using the lists you created before

In [21]:
#list includes all BMIs in order of FM1 to FM5

BMI = [weight[0]*703/(height[0]**2),
      weight[1]*703/(height[1]**2),
      weight[2]*703/(height[2]**2),
      weight[3]*703/(height[3]**2),
      weight[4]*703/(height[4]**2)]


In [22]:
print(BMI)

[21.630769230769232, 23.490755179327245, 32.99795918367347, 27.4609375, 28.36273221992869]


**E.2.3** Look up the weight classification of each person based on their BMI (what's a good source for this, who do you trust?). Create a new list with this information for each person. (Don't judge, after 9 mo of COVID we all ate too much comfort foods.)

In [23]:
BMI1 = ["normal weight","normal weight","obese","overweight","overweight"]

**E.2.3** Let's practice some print statements and iterate manually through the people on your lists.

For each person, print out something like: "First_name last_name, age xx, has a height of xx units and weight of xx units, giving them a body mass index (BMI) of xx, making them yy" where yy is if they are normal, obese, etc.

In [24]:
print(str(firstname[0])+str(lastname[0])+", age"+str(age[0])+", has a height of "+ str(height[0])+" inches and a weight of "+ str(weight[0])+" pounds, giving them a BMI of "+str(BMI[0])+", making them "+ str(BMI1[0]))

MarcelaGranados, age18, has a height of 65 inches and a weight of 130 pounds, giving them a BMI of 21.630769230769232, making them normal weight


In [25]:
print(str(firstname[1])+str(lastname[1])+", age"+str(age[1])+", has a height of "+ str(height[1])+" inches and a weight of "+ str(weight[1])+" pounds, giving them a BMI of "+str(BMI[1])+", making them "+ str(BMI1[1]))

MargothBarrios, age44, has a height of 67 inches and a weight of 150 pounds, giving them a BMI of 23.490755179327245, making them normal weight


In [26]:
print(str(firstname[2])+str(lastname[2])+", age"+str(age[2])+", has a height of "+ str(height[2])+" inches and a weight of "+ str(weight[2])+" pounds, giving them a BMI of "+str(BMI[2])+", making them "+ str(BMI1[2]))

JuanGranados, age47, has a height of 70 inches and a weight of 230 pounds, giving them a BMI of 32.99795918367347, making them obese


In [27]:
print(str(firstname[3])+str(lastname[3])+", age"+str(age[3])+", has a height of "+ str(height[3])+" inches and a weight of "+ str(weight[3])+" pounds, giving them a BMI of "+str(BMI[3])+", making them "+ str(BMI1[3]))

KarinaBarrios, age40, has a height of 64 inches and a weight of 160 pounds, giving them a BMI of 27.4609375, making them overweight


In [28]:
print(str(firstname[4])+str(lastname[4])+", age"+str(age[4])+", has a height of "+ str(height[4])+" inches and a weight of "+ str(weight[4])+" pounds, giving them a BMI of "+str(BMI[4])+", making them "+ str(BMI1[4]))

FeluelBarrios, age42, has a height of 73 inches and a weight of 215 pounds, giving them a BMI of 28.36273221992869, making them overweight


**E.2.4** Now add/append your information to each of the lists, and do the same print statement for you. Make a resolution to eat more veggies.

In [29]:
firstname.append("Paola")
lastname.append("Granados")
age.append(21)
height.append(64)
weight.append(125)

In [30]:
BMI = BMI +[weight[5]*703/(height[5]**2)]
print(BMI)

[21.630769230769232, 23.490755179327245, 32.99795918367347, 27.4609375, 28.36273221992869, 21.453857421875]


In [31]:
BMI1.append("normal weight")

In [32]:
print(str(firstname[5])+str(lastname[5])+", age"+str(age[5])+", has a height of "+ str(height[5])+" inches and a weight of "+ str(weight[5])+" pounds, giving them a BMI of "+str(BMI[5])+", making them "+ str(BMI1[5]))

PaolaGranados, age21, has a height of 64 inches and a weight of 125 pounds, giving them a BMI of 21.453857421875, making them normal weight


When you are done with your lab, upload this document to your github folder on the class site. Then submit a url link to the completed lab, along with a screenshot of your completed Ch 2 score in datacamp, to the assignment tab in Blackboard.

# Scratch

In [5]:
# this is an area of the notebook you can mess around and test things out, if you don't want to put it above

In [36]:
print(lastname)

['Granados', 'Barrios', 'Granados', 'Barrios', 'Barrios', 'Granados']
